# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mp4/MP4Demuxer.cpp
# Commit: 89c241451a32
# Full Hash: 89c241451a32d362967315a0e2424fcce824208b
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-05-18 03:51:42
# Regressor Bug: 1817997
# File Overlap Count: 1
# Description:
#   Bug 1817997 - Update the MP4 demuxer to use TimeUnit based on the internal MP4 time base. r=alwu
#   
#   This still gets the initial time value from mp4parse-rust, that is in
#   microseconds. mp4parse-rust has been updated to expose real time, and will be
#   updated later.
# ==============================================================================

diff -r 66e4a26e1ebe -r 89c241451a32 dom/media/mp4/MP4Demuxer.cpp
--- a/dom/media/mp4/MP4Demuxer.cpp	Wed May 17 15:47:04 2023 +0000
+++ b/dom/media/mp4/MP4Demuxer.cpp	Wed May 17 15:47:04 2023 +0000
@@ -356,7 +356,7 @@
   auto seekTime = aTime;
   mQueuedSample = nullptr;
 
-  mIterator->Seek(seekTime.ToMicroseconds());
+  mIterator->Seek(seekTime);
 
   // Check what time we actually seeked to.
   do {
@@ -480,16 +480,16 @@
 
 void MP4TrackDemuxer::SetNextKeyFrameTime() {
   mNextKeyframeTime.reset();
-  Microseconds frameTime = mIterator->GetNextKeyframeTime();
-  if (frameTime != -1) {
-    mNextKeyframeTime.emplace(media::TimeUnit::FromMicroseconds(frameTime));
+  media::TimeUnit frameTime = mIterator->GetNextKeyframeTime();
+  if (frameTime.IsValid()) {
+    mNextKeyframeTime.emplace(frameTime);
   }
 }
 
 void MP4TrackDemuxer::Reset() {
   mQueuedSample = nullptr;
-  // TODO, Seek to first frame available, which isn't always 0.
-  mIterator->Seek(0);
+  // TODO: verify this
+  mIterator->Seek(media::TimeUnit::FromNegativeInfinity());
   SetNextKeyFrameTime();
 }
 